src/cljs/athens/events.cljs

```diff
@@ -338,6 +338,18 @@
          :dispatch [:page/create title uid]}))))
 
 
+(reg-event-fx
+  :daily-note/delete
+  (fn [{:keys [db]} [_ uid]]
+    (let [retract-blocks (db/retract-uid-recursively uid)
+          tx-data        (if (db/e-by-av :block/uid uid)  ;; If a page for the Daily Note exists
+                           retract-blocks                 ;; retract both the page and the child blocks
+                           (rest retract-blocks))         ;; else, only retract the child blocks
```

Comment:
Not sure I understand this `if` statement. How could a Daily Note page exist without children? 

Second, why not just dispatch `:page/delete` from within `:daily-note/delete`? All the code from `:page/delete` can be reused. For instance, looking at `:page/delete`, I see there is also a condition that checks for Linked Refs, that is not being checked here.

Reply:
I was trying to workaround this issue that popped up when dispatching `:page/delete` on the Daily Notes page. 

`"EXCEPTION" #error {:message "Nothing found for entity id [:block/uid \"02-16-2021\"]", :data {:error :entity-id/missing, :entity-id [:block/uid "02-16-2021"]}}`.

The only way I could get it to work was to remove the `[:db/retractEntity [:block/uid "02-16-2021"]]` item from the `retract-blocks` vector and pass that on to `tx-data`

Good catch on the linked refs part. I've fixed that now.